sympy/solvers/bivariate.py

```diff
@@ -176,22 +185,73 @@ def _lambert(eq, x):
     p, den = den.as_coeff_Mul()
     e = exp(num/den)
     t = Dummy('t')
-    args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
-
     # calculating solutions from args
-    for arg in args:
+    root_list = list(roots(t**p - e, t))
+    if domain is S.Reals:
+        root_list = [r for r in root_list if not r.has(I)]
+    args = [r*d/(a*b) for r in root_list]
+    if len(args) == 0:
+        return S.EmptySet
+    # for simplicity defined two functions
+    # Lambert_Real_Braches , lambert_helper
+    def lambert_braches():
         for k in lambert_real_branches:
-            w = LambertW(arg, k)
-            if k and not w.is_real:
+            rhs2 = -c/b + (a/d)*LambertW(arg, k)
+            if rhs2.has(I):
                 continue
-            rhs = -c/b + (a/d)*w
+            sol1 = lambert_helper(rhs2)
+        return sol1, rhs2
+
+    def lambert_helper(rhs):
+    # to avoid repetitive code
+        for xu in xusolns:
+            sol.append(xu.subs(u, rhs))
+        return sol
+
+    n = Dummy('n',integer=True)
+    len_of_symbol = len(list(eq.atoms(Symbol,Dummy)))
+    for arg in args:
+        rhs = -c/b + (a/d)*LambertW(arg)
+        lambert_real = LambertW(arg).is_real
+
+        if domain_real:
+            if not len_of_symbol >1:
```

Comment:
There should be a space after an operator like `>` but also it would make more sense to express this as
```
if len_of_symbol <= 1:
```
or whatever it is supposed to be.